.github/workflows/release.yaml

```diff
@@ -23,23 +23,19 @@ jobs:
       fail-fast: false
       matrix:
         runs-on: [[ubuntu-22.04], [self-hosted, jammy, ARM64]]
-    outputs:
-      snap: ${{ steps.build.outputs.snap }}
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0  # Complete git history is required to generate the version from git tags.
-      - uses: canonical/action-build@v1
-        id: build
       - name: Determine system architecture
         run: echo "SYSTEM_ARCH=$(uname -m)" >> $GITHUB_ENV
-      - uses: actions/upload-artifact@v4
+      - name: Download the built snap from check workflow
+        uses: actions/download-artifact@v4
         with:
           name: snap_${{ env.SYSTEM_ARCH }}
-          path: ${{ steps.build.outputs.snap }}
       - uses: snapcore/action-publish@v1
         env:
           SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
         with:
-          snap: ${{ steps.build.outputs.snap }}
+          snap: snap_${{ env.SYSTEM_ARCH }}
```

Comment:
This is just the artefact ID, not the file name.  You'll need to use some logic to find the file name  - eg. a step like (untested):

```
echo "SNAP_FILE=$(find . -name "*.snap")" >> $GITHUB_ENV
```

Reply:
You can use the name here, we did the same in `check.yaml`.

Reply:
I don't think so - `check.yaml` doesn't do anything more than download the artefact.  The func tests use a glob to find the snap file name https://github.com/canonical/dcgm-snap/blob/4c2c77f09eb475ce863132879fe15940908783ec/tests/functional/conftest.py#L9

Reply:
Ah, I see what you mean, you're right, there should be a filename glob or something